samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/qti/XMLImportBean.java

```diff
@@ -462,70 +464,117 @@ public void setAuthorizationBean(AuthorizationBean authorizationBean)
    * Value change on upload
    * @param e the event
    */
-  public void importPoolFromQti(ValueChangeEvent e)
-  {
+  public void importPool(ValueChangeEvent e) {
+    String sourceType = ContextUtil.lookupParam("sourceType");
     String uploadFile = (String) e.getNewValue();
 
-    try
-    {
-    	processAsPoolFile(uploadFile);
+    if (uploadFile!= null && uploadFile.startsWith("SizeTooBig:")) {
+        Long sizeMax = Long.valueOf(ServerConfigurationService.getString("samigo.sizeMax", "20480"));
+        String sizeTooBigMessage = MessageFormat.format(rb.getString("import_size_too_big"), uploadFile.substring(11), Math.round(sizeMax.floatValue()/1024));
+        FacesMessage message = new FacesMessage(sizeTooBigMessage);
+        FacesContext.getCurrentInstance().addMessage(null, message);
+        // remove unsuccessful file
+        log.debug("****Clean up file:"+uploadFile);
+        File upload = new File(uploadFile);
+        upload.delete();
+        authorBean.setImportOutcome("importPool");
+        return;
     }
-    catch (Exception ex)
-    {
-      FacesMessage message = new FacesMessage( rb.getString("import_err_pool") );
-      FacesContext.getCurrentInstance().addMessage(null, message);
+    authorBean.setImportOutcome("author");
+
+    if ("2".equals(sourceType)) {
+        if(uploadFile.toLowerCase().endsWith(".zip")) {
+            isCP = true;
+            importPool(uploadFile, true, true);
+        } else {
+            isCP = false;
+            importPool(uploadFile, false, true);
+        }
+    } else {
+         if(uploadFile.toLowerCase().endsWith(".zip")) {
+             isCP = true;
+             importPool(uploadFile,true, false);
+         } else {
+             isCP = false;
+             importPool(uploadFile, false, false);
+         }
     }
   }
-   
-  
+
   /**
    * Process uploaded QTI XML 
    * assessment as question pool
- * @throws Exception 
    */
-  private void processAsPoolFile(String uploadFile) throws Exception
-  {
-    itemAuthorBean.setTarget(ItemAuthorBean.FROM_QUESTIONPOOL); // save to questionpool
+  private void importPool(String uploadFile, boolean isCP, boolean isRespondus) {
 
     // Get the file name
     String fileName = uploadFile;
+    String unzipLocation = null;
+    boolean fileNotFound = false;
+    if (isCP) {
+        ImportService importService = new ImportService();
+        unzipLocation = importService.unzipImportFile(uploadFile);
+        fileName = unzipLocation + "/" + importService.getQtiFilename();
```

Comment:
Shall this be File.Separator instead of "/" ?

Reply:
Are you coding on Windows or Unix?

Reply:
Windows

Reply:
And it works? nice :D Is usually problematic